core/src/test/java/org/elasticsearch/action/search/SearchResponseTests.java

```diff
@@ -0,0 +1,174 @@
+            failures[i] = ShardSearchFailureTests.createTestItem();
+        }
+        SearchResponse response = createTestItem(failures);
+        XContentType xcontentType = randomFrom(XContentType.values());
+        final ToXContent.Params params = new ToXContent.MapParams(singletonMap(RestSearchAction.TYPED_KEYS_PARAM, "true"));
+        BytesReference originalBytes = toShuffledXContent(response, xcontentType, params, randomBoolean());
+        try (XContentParser parser = createParser(xcontentType.xContent(), originalBytes)) {
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
+            SearchResponse parsed = SearchResponse.fromXContent(parser);
+            for (int i = 0; i < parsed.getShardFailures().length; i++) {
+                ShardSearchFailure parsedFailure = parsed.getShardFailures()[i];
+                ShardSearchFailure originalFailure = failures[i];
+                assertEquals(originalFailure.index(), parsedFailure.index());
+                assertEquals(originalFailure.shard().getNodeId(), parsedFailure.shard().getNodeId());
+                assertEquals(originalFailure.shardId(), parsedFailure.shardId());
+                String originalMsg = originalFailure.getCause().getMessage();
+                assertEquals(parsedFailure.getCause().getMessage(), "Elasticsearch exception [type=parsing_exception, reason=" +
+                        originalMsg + "]");
+                String nestedMsg = originalFailure.getCause().getCause().getMessage();
+                assertEquals(parsedFailure.getCause().getCause().getMessage(),
+                        "Elasticsearch exception [type=illegal_argument_exception, reason=" + nestedMsg + "]");
+            }
+            assertEquals(XContentParser.Token.END_OBJECT, parser.currentToken());
+            assertNull(parser.nextToken());
+        }
+    }
+
+    public void testToXContent() {
+        SearchHit hit = new SearchHit(1, "id1", new Text("type"), Collections.emptyMap());
+        hit.score(2.0f);
+        SearchHit[] hits = new SearchHit[] { hit };
+        SearchResponse response = new SearchResponse(
+                new InternalSearchResponse(new SearchHits(hits, 100, 1.5f), null, null, null, false, null, 1), null, 0, 0, 0,
+                new ShardSearchFailure[0]);
+        assertEquals(
+                "{\"took\":0,"
+                        + "\"timed_out\":false,"
+                        + "\"_shards\":"
+                        + "{\"total\":0,"
+                        + "\"successful\":0,"
+                        + "\"failed\":0"
+                        + "},"
+                        + "\"hits\":"
+                        + "{\"total\":100,"
+                        + "\"max_score\":1.5,"
+                        + "\"hits\":[{\"_type\":\"type\",\"_id\":\"id1\",\"_score\":2.0}]"
+                        + "}"
+                        + "}", Strings.toString(response));
```

Comment:
Although I liked it at first, I'm starting to doubt the usefulness of these tests. And formatting makes it unreadable. If we keep it, can you change this to something like:

```
StringBuilder expectedString = new StringBuilder();
        expectedString.append("{");
        {
            expectedString.append("\"took\":0,");
            expectedString.append("\"timed_out\":false,");
            expectedString.append("\"_shards\":");
            {
                expectedString.append("{\"total\":0,");
                expectedString.append("\"successful\":0,");
                expectedString.append("\"failed\":0},");
            }
            expectedString.append("\"hits\":");
            {
                expectedString.append("{\"total\":100,");
                expectedString.append("\"max_score\":1.5,");
                expectedString.append("\"hits\":[{\"_type\":\"type\",\"_id\":\"id1\",\"_score\":2.0}]}");
            }
        }
        expectedString.append("}");
        assertEquals(expectedString.toString(), Strings.toString(response));
```
That should be more robust against auto-formatting.